File: test-api.py
#!/usr/bin/env python3

"""
Demo script to test JWT authentication and models API
Usage: python3 test-api.py
"""

import requests
import json

BASE_URL = "http://localhost:3000"

def test_api():
    print("🧪 Testing WejdanAI JWT Authentication and Models API")
    print(f"📡 Base URL: {BASE_URL}")
    
    try:
        # Step 1: Generate a JWT token
        print("\n1️⃣ Generating JWT token...")
        
        auth_response = requests.post(f"{BASE_URL}/api/auth", 
            json={"username": "testuser"},
            headers={"Content-Type": "application/json"}
        )
        
        auth_data = auth_response.json()
        print("✅ Auth response:", auth_data)
        
        if "token" not in auth_data:
            raise Exception("Failed to get token")
        
        token = auth_data["token"]
        
        # Step 2: Test accessing models API without token (should fail)
        print("\n2️⃣ Testing models API without token (should fail)...")
        
        try:
            no_auth_response = requests.get(f"{BASE_URL}/api/models-demo")
            print("🚫 Status:", no_auth_response.status_code)
            print("🚫 Response:", no_auth_response.json())
        except Exception as e:
            print("🚫 Request failed as expected:", str(e))
        
        # Step 3: Add a model using JWT token
        print("\n3️⃣ Adding a model with JWT token...")
        
        add_model_response = requests.post(f"{BASE_URL}/api/models-demo",
            json={
                "name": "GPT-4",
                "version": "4.0", 
                "description": "OpenAI model for testing"
            },
            headers={
                "Content-Type": "application/json",
                "Authorization": f"Bearer {token}"
            }
        )
        
        add_model_data = add_model_response.json()
        print("✅ Add model response:", add_model_data)
        
        # Step 4: Fetch all models
        print("\n4️⃣ Fetching all models...")
        
        get_models_response = requests.get(f"{BASE_URL}/api/models-demo",
            headers={"Authorization": f"Bearer {token}"}
        )
        
        models_data = get_models_response.json()
        print("✅ Models list:", models_data)
        
        print("\n🎉 API test completed successfully!")
        
    except Exception as error:
        print(f"❌ Test failed: {error}")
        return False
    
    return True

if __name__ == "__main__":
    test_api()

File: test-api.js
#!/usr/bin/env node

// Demo script to test JWT authentication and models API
// Run with: node test-api.js

const BASE_URL = process.env.BASE_URL || "http://localhost:3000"

async function testAPI() {
  console.log("🧪 Testing WejdanAI JWT Authentication and Models API")
  console.log(`📡 Base URL: ${BASE_URL}`)
  
  try {
    // Step 1: Generate a JWT token
    console.log("\n1️⃣ Generating JWT token...")
    
    const authResponse = await fetch(`${BASE_URL}/api/auth`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ username: 'testuser' })
    })
    
    const authData = await authResponse.json()
    console.log("✅ Auth response:", authData)
    
    if (!authData.token) {
      throw new Error("Failed to get token")
    }
    
    const token = authData.token
    
    // Step 2: Test accessing models API without token (should fail)
    console.log("\n2️⃣ Testing models API without token (should fail)...")
    
    try {
      const noAuthResponse = await fetch(`${BASE_URL}/api/models`)
      const noAuthData = await noAuthResponse.json()
      console.log("🚫 Expected unauthorized response:", noAuthData)
    } catch (error) {
      console.log("🚫 Request failed as expected:", error.message)
    }
    
    // Step 3: Add a model using JWT token
    console.log("\n3️⃣ Adding a model with JWT token...")
    
    const addModelResponse = await fetch(`${BASE_URL}/api/models`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`
      },
      body: JSON.stringify({
        name: 'GPT-4',
        version: '4.0',
        description: 'OpenAI model for testing'
      })
    })
    
    const addModelData = await addModelResponse.json()
    console.log("✅ Add model response:", addModelData)
    
    // Step 4: Fetch all models
    console.log("\n4️⃣ Fetching all models...")
    
    const getModelsResponse = await fetch(`${BASE_URL}/api/models`, {
      headers: {
        'Authorization': `Bearer ${token}`
      }
    })
    
    const modelsData = await getModelsResponse.json()
    console.log("✅ Models list:", modelsData)
    
    console.log("\n🎉 API test completed successfully!")
    
  } catch (error) {
    console.error("❌ Test failed:", error.message)
    process.exit(1)
  }
}

// Run the test
testAPI()